### ISSUE TITLE
conda install UnsatisfiableError python versions

### ISSUE THREAD

**Issue by Firedrops** [ablab/quast#112]
I am attempting to install quast in a docker image through conda
`conda install -y -c bioconda quast`

I get this error when trying to build the image:
```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - quast -> python[version='2.7.*|>=2.7,<2.8.0a0|>=3.5,<3.6.0a0|>=3.6,<3.7.0a0']

Your python: python=3.7

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

The following specifications were found to be incompatible with each other:



Package pip conflicts for:
quast -> python=2.7 -> pip
python=3.7 -> pip
Package ca-certificates conflicts for:
quast -> python=2.7 -> ca-certificates
python=3.7 -> openssl[version='>=1.0.2o,<1.0.3a'] -> ca-certificates
Package wheel conflicts for:
quast -> python=2.7 -> pip -> wheel
python=3.7 -> pip -> wheel
Package certifi conflicts for:
quast -> matplotlib -> setuptools -> certifi[version='>=2016.09']
python=3.7 -> pip -> setuptools -> certifi[version='>=2016.09']
Package python-dateutil conflicts for:
quast -> simplejson -> python-dateutil
Package setuptools conflicts for:
python=3.7 -> pip -> setuptools
quast -> matplotlib -> setuptools
```

The docker is based on Debian 9 and has both `python-pip` and `python3-pip` installed.

What am I doing wrong?

Edit: [Dockerfile code here](https://github.com/Firedrops/qimr_mtb/blob/master/Dockerfile), quast part at line 111.

**Comment by iquasere** [comment]
I came here searching an answer for the same problem. You (we) are doing nothing wrong. As the error message explains, we are trying to install quast on a conda environment that has python 3.7 installed. As far as I was able to tell, there is no QUAST distribution available in conda for python 3.7. The solution I have found was to install it via pip.
```
pip install quast
```
A robust solution for this problem, and also a good practice for docker image builders, would be to specifiy the versions on all the tools, so none installs python > 3.6.8 as prerequisite.

**Comment by qiutaoding** [comment]
You can create a new environment with python 3.6
 `conda create -n quast-env python=3.6`
`conda activate quast-env`
`conda install -c bioconda quast`
`quast --help`
Then it  works.

**Comment by iquasere** [comment]
It doesn't work if you want to use it in a docker image. Pip install is the easiest solution in this case, and works as good as using environments, but with less layers of container and error. Now, if you **really** want to install quast with conda, you can use the ENV hability of Dockerfile:
```
RUN conda create -n py36 python=3.6
RUN echo "source activate py36" > ~/.bashrc
ENV PATH /opt/conda/envs/env/bin:$PATH
CMD [ " conda activate py36 && conda install -c bioconda quast" ]
```
and then you have to set your code so it activates that environment every time you want to use quast.

**Comment by hafez-ahmad** [comment]
is keras not available for python 3.8 ? 
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - keras -> python[version='>=3.5,<3.6.0a0|>=3.6,<3.7.0a0']

Your python: python=3.8

If python is on the left-most side of the chain, that's the version you've asked for.
When python appears to the right, that indicates that the thing on the left is somehow
not available for the python version you are constrained to. Note that conda will not
change your python version to a different minor version unless you explicitly specify
that.

**Comment by rabiul-ai** [comment]
I also fallen into the same problem for Python 3.9. Later just I use 
`pip install -U scikit-learn`
then it works. Thanks.
